### PR TITLE
fix: move tests into lib

### LINKED ISSUE #5606: Move tests into lib
Tests for datasources and managers are already part of `lib`. Now it's time to move the rest.

### PR DESCRIPTION
**Pull request by ViralRuparel**
Moves all the test files from test folder to lib folder

Closes #5606

### CODE REVIEW

**Comment by ViralRuparel** [comment]
Ready for review @rarkins

File: lib/platform/azure/azure-helper.spec.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ describe('platform/azure/helpers', () => {
     // reset module
     jest.resetModules();
     jest.mock('../../../lib/platform/azure/azure-got-wrapper');
-    azureHelper = require('../../../lib/platform/azure/azure-helper');
+    azureHelper = require('.../../../lib/platform/azure/azure-helper');
```
Comment (rarkins) [inline lib/platform/azure/azure-helper.spec.ts]:
```suggestion
    azureHelper = require('./azure-helper');
```

File: lib/platform/bitbucket-server/__fixtures__/responses.ts [hunk 2]
```diff
@@ -179,7 +179,7 @@ function generateServerResponses(endpoint) {
     [`${endpoint}/rest/api/1.0/projects/SOME/repos/repo/pull-requests/5/commits?withCounts=true`]: {
       GET: {
         totalCount: 1,
-        values: [ { author: { emailAddress: 'bot@renovateapp.com'} } ],
+        values: [{ author: { emailAddress: 'bot@renovateapp.com' } }],
```
Comment (rarkins) [inline lib/platform/bitbucket-server/__fixtures__/responses.ts]: @viceice on one hand I was going to say "Revert formatting changes to fixtures" but on the other hand I wonder if this file is an indicator that prettier is picking up *.ts files within `__fixtures__` and needs reconfiguring?

File: lib/platform/gitea/gitea-got-wrapper.spec.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
-import { GotResponse } from '../../../lib/platform';
-import { partial } from '../../util';
-import { GotFn } from '../../../lib/util/got';
-import { GiteaGotApi } from '../../../lib/platform/gitea/gitea-got-wrapper';
+import { GotResponse } from '..';
+import { partial } from '../../../test/util';
```
Comment (rarkins) [inline lib/platform/gitea/gitea-got-wrapper.spec.ts]: Need to work out what to do about this util file

File: lib/versioning/poetry/poetry.spec.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { api as semver } from '../../lib/versioning/poetry';
+import { api as semver } from '.';
```
Comment (rarkins) [inline lib/versioning/poetry/poetry.spec.ts]: I think we should rename these types of files to `index.spec.ts` now that they're within `lib`. Please check all versioning, manager and datasource ones

**Review by rarkins — CHANGES_REQUESTED** [review]

**Comment by rarkins** [comment]
@viceice any ideas on what to do about the test/util code that remains?

**Comment by rarkins** [comment]
This is how the `test/` directory looks now:

image

I guess it's ok to leave as-is. I can't think of any better alternatives.

For instance, I thought about moving the docs test into docs/ but that could be "messy" for someone browsing the repo just for the docs and not source code.

File: lib/versioning/poetry/poetry.spec.ts (hunk 4 above)
Comment (ViralRuparel) [reply lib/versioning/poetry/poetry.spec.ts]: okay will change the names

File: lib/platform/gitea/gitea-got-wrapper.spec.ts (hunk 3 above)
Comment (ViralRuparel) [reply lib/platform/gitea/gitea-got-wrapper.spec.ts]: yes please suggest, until then keeping as is.

File: lib/platform/azure/azure-helper.spec.ts (hunk 1 above)
Comment (ViralRuparel) [reply lib/platform/azure/azure-helper.spec.ts]: done everywhere in all the test files.

File: lib/versioning/poetry/poetry.spec.ts (hunk 4 above)
Comment (ViralRuparel) [reply lib/versioning/poetry/poetry.spec.ts]: Done with this too.

**Comment by ViralRuparel** [comment]
Again ready for review @rarkins with all the suggested changes and renaming of test files done.

**Pull request merged**